packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemChildren.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { hasSubmenu, getIsChecked } from '../../utilities/contextMenu';
+import { IContextualMenuItem } from './ContextualMenu.types';
+import { IMenuItemClassNames } from './ContextualMenu.classNames';
+import { getRTL } from '../../Utilities';
+import { Icon, IIconProps } from '../../Icon';
+import { IContextualMenuItemChildrenProps } from './ContextualMenuItemChildren.types';
+
+const renderIcon = ({ hasIcons, item, classNames }: IContextualMenuItemChildrenProps) => {
```

Comment:
Let's capitalize all of these and use them as JSX elements. It's minor but I think it keeps us moving in the direction of thinking in more JSX element type ways.

Reply:
I really like that idea. We don't do that in our front end repo, and we should. Going to start always doing this. Thanks Jon

Reply:
Hmmm so there is a bit of a limitation with enzyme here. I can no longer use ShallowWrapper to tests these once they are react components. I have to do a full mount which really blows up the PR size and makes snapshots impossible to parse. I think there are a couple options. 

1. Separate out all those little subcomponents into their own files with their own tests. Seems really verbose for tiny little things that no other part of the codebase cares about.
2. Full mount and huge snapshots.
3. Return to functions and ShallowWrapper. Makes for human readable snapshots and less verbose, but looks less JSXy.

What do you think @joschect?

Reply:
@micahgodbolt will probably want to weigh in as well. I don't feel comfortable with the size of the PR due to the snapshots right now

Reply:
That's tough. I'd say separating them out into separate files isn't great. I'd like to keep them as jsx elements but if that's not possible I think it would be okay to go back to functions.

Reply:
I tried keeping the mount and mocking out the <Icon /> children components that they render. Unfortunately that didn't cut much off the bundle size. Sadly I endedup going back to the functions, which saved around 2500 lines off the snapshots.